processor.go

```diff
@@ -119,38 +119,16 @@ func (a *Processor) Check(uaFingerprint fp.UAFingerprint, rawUa string,
 	}
 
 	// Remove grease ciphers, extensions, and curves from request fingerprint and add as quirk instead.
-	hasGrease := false
-	idx := 0
-	for _, elem := range actualReqFin.Cipher {
-		if (elem & 0x0f0f) == 0x0a0a {
-			hasGrease = true
-		} else {
-			actualReqFin.Cipher[idx] = elem
-			idx++
-		}
-	}
-	actualReqFin.Cipher = actualReqFin.Cipher[:idx]
-	idx = 0
-	for _, elem := range actualReqFin.Extension {
-		if (elem & 0x0f0f) == 0x0a0a {
-			hasGrease = true
-		} else {
-			actualReqFin.Extension[idx] = elem
-			idx++
-		}
-	}
-	actualReqFin.Extension = actualReqFin.Extension[:idx]
-	idx = 0
-	for _, elem := range actualReqFin.Curve {
-		if (elem & 0x0f0f) == 0x0a0a {
-			hasGrease = true
-		} else {
-			actualReqFin.Curve[idx] = elem
-			idx++
-		}
-	}
-	actualReqFin.Curve = actualReqFin.Curve[:idx]
-	if hasGrease {
+	hasGreaseCipher, greaseBound := detectGrease(actualReqFin.Cipher)
+	actualReqFin.Cipher = actualReqFin.Cipher[:greaseBound] // Remove grease ciphers
```

Comment:
I would perhaps change the name of the `detectGrease` function to `removeGrease` instead, since the function is actually detecting _and removing_ the grease ciphers from the list in the loop iteration. Then, maybe changing the name of the `greaseBound` variable to something like `newSize` would be a bit more clear, since that's the number of elements left in the list after grease ciphers have been skipped over. Would you agree?

Reply:
I agree! I'll do that right now.